Sources/MapboxMaps/Viewport/Helpers/MainQueue.swift

```diff
@@ -1,11 +0,0 @@
-// depending on this protocol instead of on DispatchQueue directly
-// allow mocking the main queue in tests which avoids the need for waits
-internal protocol MainQueueProtocol: AnyObject {
-    func async(execute work: @escaping () -> Void)
-}
-
-internal final class MainQueue: MainQueueProtocol {
-    internal func async(execute work: @escaping () -> Void) {
-        DispatchQueue.main.async(execute: work)
-    }
-}
```

Comment:
Not sure if enforcing particular instances with protocols brings much benefits - `DispatchQueue.main` is an instance of `DispatchQueue` after all. Even this approach with `MainQueueProtocol` never enforced anything - nothing prevented somebody to define another conformant(of the `MainQueueProtocol`) with whichever dispatch queue underneath and pass it to a place that accepts `MainQueueProtocol`.
I think things like this could be checked in unit tests.